docs/book/src/developer/e2e.md

```diff
@@ -0,0 +1,226 @@
+  wait commands/for the `Eventually` method.
+- Define the list of images to be loaded in the management cluster (this is specif of
+  management cluster based on kind).
+
+An [example E2E config file] can be found here.
+
+<aside class="note">
+
+<h1>Deprecated E2E config file format</h1>
+
+The [Cluster API test framework] includes also a [deprecated E2E config file] implementation,
+that was used before the introduction of clusterctl. . This might be removed in future releases 
+of the test framework.
+
+</aside>
+
+### Creating the management cluster and installing providers
+
+The [InitManagementCluster method] provides a convenient way for creating a management cluster
+and installing providers.
+
+This method:
+- Creates a management cluster; you can provide your own create func or use one of the 
+  existing create functions: [NewCluster] and [NewClusterForCAPD].
+- If supported by the management cluster type, it loads images on the management cluster,
+  as defined in the [E2E config file].
+- Creates a clusterctl local repository with all the providers defined in the [E2E config file].
+- Runs `clusterctl init` using the above local repository.
+- Waits for the providers controllers to be running.
+
+<aside class="note">
+
+<h1>Deprecated InitManagementCluster method</h1>
+
+The [Cluster API test framework] includes also a [deprecated InitManagementCluster method] implementation,
+that was used before the introduction of clusterctl. This might be removed in future releases 
+of the test framework.
+
+</aside>
+
+## Writing test specs
+
+A typical test spec is a sequence of:
+
+- Creating objects in the management cluster, wait for the corresponding infrastructure to be provisioned.
+- Exec clusterctl operations like e.g. `clusterctl move`, wait for the action to complete.
+- Delete objects in the management cluster, wait for the corresponding infrastructure to be terminated.
+
+### Creating objects
+
+There are two possible approaches for creating objects in the management cluster:
+
+- Create object by object: create the `Cluster` object, then `AwsCluster`, `Machines`, `AwsMachines` etc.
+- Apply a `cluster-templates.yaml` file thus creating all the objects this file contains. 
+
+The first approaches leverage on the [controller-runtimeClient] and gives you full control, but it comes with
+some drawbacks as well, because this method does not reflec directly real user workflows, and most important
+the resulting tests are hardly reusable with different infrastructure providers (see [writing portable tests](#writing_portable_test)).
+
+As alternative approach, you can use the [ClusterTemplate method] and the [Apply method] for creating objects in the cluster.
+This methods mimics the recommended user workflows, and it based on `cluster-templates.yaml` files that can be
+provided via the [E2E config file], and thus easily swappable when changing the target infrastructure provider.  
```

Comment:
It seems we want the developer to lean towards this method of writing e2e tests. If so I suggest,
```suggestion
We recommend using the [ClusterTemplate method] and the [Apply method] for creating objects in the cluster.
This methods mimics the recommended user workflows, and it is based on `cluster-templates.yaml` files that can be
provided via the [E2E config file], and thus easily swappable when changing the target infrastructure provider.  
```